include/RAJA/pattern/params/reducer.hpp

```diff
@@ -83,16 +55,41 @@ namespace detail
   // Basic Reducer
   //
   //
-  template <typename Op, typename T>
+
+  // Basic data type Reducer
+  // T must be a basic data type
+  // VOp must be ValOp<T, Op>
+  template <typename Op, typename T, typename VOp>
   struct Reducer : public ForallParamBase {
     using op = Op;
-    using value_type = T;
+    using value_type = T; // This is a basic data type
 
-    RAJA_HOST_DEVICE Reducer() {}
-    Reducer(value_type *target_in) : target(target_in), val(op::identity()) {}
+    Reducer() = default;
 
+    // Basic data type constructor
+    RAJA_HOST_DEVICE Reducer(value_type *target_in) : m_valop(VOp{}), target(target_in){}
```

Comment:
Here m_valop gets initialized to its default, which may be wrong if it is used as that may not be the identity, but below it gets initialized to target which may be wrong if its used more than once. Which is the right behavior?

Reply:
It looks like this value always gets overwritten before its used, should it just always be default initialized or the identity then?

Reply:
The current initialization works because we're only expecting a basic data type, and the `m_valop` is copying its final value to the target in the end. The initialization for the partial specialization below uses the target to initialize the `m_valop` because we are getting either a reference to a `ValLoc`, or references to the actual data and location.

I don't quite understand how an identity for a basic data type would be wrong because `VOp` knows the basic data type, and I'm not sure how a `ValLoc` initialization to the target could be wrong if it's used more than once. Are you saying that neither of these initializations can work, and that we need yet another way to initialize?

Reply:
As agreed, the constructor for the partial specialization now matches this one.